bal_tools/pools_gauges.py

```diff
@@ -116,33 +116,38 @@ def query_root_gauges(self, skip=0, step_size=100) -> list:
             result += self.query_root_gauges(skip + step_size, step_size)
         return result
 
-    def query_all_gauges(self, include_other_gauges=True) -> list:
+    def query_all_pools(self) -> List[PoolData]:
+        """
+        query all pools from the apiv3 subgraph
+        filters out disabled pools
+        """
+        data = self.subgraph.fetch_graphql_data("apiv3", "get_pools", {"chain": self.chain.upper()})
+        pools = [PoolData(**flatten_nested_dict(pool)) for pool in data["poolGetPools"]]
+        return [
+            pool for pool in pools
+            if pool.dynamicData.swapEnabled
+        ]
+
+    def query_all_gauges(self, include_other_gauges=True) -> List[GaugePoolData]:
         """
         query all gauges from the apiv3 subgraph
         """
         data = self.subgraph.fetch_graphql_data("apiv3", "get_gauges", {"chain": self.chain.upper()})
         all_gauges = []
         for gauge in data["poolGetPools"]:
-            if gauge['staking'] is not None:
-                if gauge['staking']['gauge'] is not None:  # this is an edge case for the 80bal20weth pool
-                    all_gauges.append({"id": gauge['staking']['gauge']['gaugeAddress'], "symbol": f"{gauge['symbol']}-gauge"})
-                    if include_other_gauges:
-                        for other_gauge in gauge['staking']['gauge']['otherGauges']:
-                            all_gauges.append({"id": other_gauge['id'], "symbol": f"{gauge['symbol']}-gauge"})
+            pool = GaugePoolData(**flatten_nested_dict(gauge))
```

Comment:
hmm im not sure about adding `StakingData` and `GaugePoolData`

if i call `query_all_gauges`, i expect to get back a (rich) list of gauges. right now it returns a collection of stakings, for which i then have to go one level lower to find the gauge.

also the logic to just parse other gauges as a gauge disappeared, they now live nested in the staking object again

Reply:
i see the intention now, modified in 7ff2440: `query_all_gauges` now returns list of `GaugeData`, which has the id and symbol at top level, also fixed `include_other_gauges` logic.